registration/src/main/java/org/spine3/samples/lobby/registration/seat/availability/SeatsAvailabilityAggregate.java

```diff
@@ -0,0 +1,366 @@
+        final SeatsAvailability.Builder state = getState().toBuilder();
+
+        final Map<String, SeatQuantities> pendingReservations = state.getMutablePendingReservations();
+        final String reservationId = event.getReservationId().getUuid();
+        pendingReservations.remove(reservationId);
+
+        incrementState(state.build());
+    }
+
+    @Apply
+    private void apply(SeatsReservationCancelled event) {
+        final SeatsAvailability.Builder state = getState().toBuilder();
+
+        final Map<String, SeatQuantities> pendingReservations = state.getMutablePendingReservations();
+        final String reservationId = event.getReservationId().getUuid();
+        pendingReservations.remove(reservationId);
+
+        state.clearAvailableSeat();
+        state.addAllAvailableSeat(event.getAvailableSeatUpdatedList());
+
+        incrementState(state.build());
+    }
+
+    @Apply
+    private void apply(AddedAvailableSeats event) {
+        final SeatsAvailability.Builder state = getState().toBuilder();
+
+        final List<SeatQuantity> availableSeats = state.getAvailableSeatList();
+        final SeatQuantity addedQuantity = event.getQuantity();
+        final SeatTypeId seatTypeId = addedQuantity.getSeatTypeId();
+        final SeatQuantity existingOne = findById(availableSeats, seatTypeId, null);
+        if (existingOne != null) {
+            final int indexOfOldValue = availableSeats.indexOf(existingOne);
+            final int newQuantity = existingOne.getQuantity() + addedQuantity.getQuantity();
+            state.setAvailableSeat(indexOfOldValue, newSeatQuantity(seatTypeId, newQuantity));
+        } else {
+            state.addAvailableSeat(addedQuantity);
+        }
+
+        incrementState(state.build());
+    }
+
+    @Apply
+    private void apply(RemovedAvailableSeats event) {
+        final SeatsAvailability.Builder state = getState().toBuilder();
+
+        final SeatQuantity removedQuantity = event.getQuantity();
+        final SeatTypeId seatTypeId = removedQuantity.getSeatTypeId();
+        final List<SeatQuantity> availableSeats = state.getAvailableSeatList();
+        final SeatQuantity existingOne = findById(availableSeats, seatTypeId);
+        final int indexOfOldValue = availableSeats.indexOf(existingOne);
+        final int newQuantity = calculateNewQuantity(removedQuantity, existingOne);
+        state.setAvailableSeat(indexOfOldValue, newSeatQuantity(seatTypeId, newQuantity));
+
+        incrementState(state.build());
+    }
+
+    private static int calculateNewQuantity(SeatQuantity removedQuantity, SeatQuantity existingOne) {
+        final int newQuantity = existingOne.getQuantity() - removedQuantity.getQuantity();
+        if (newQuantity < 0) {
+            return 0;
+        }
+        return newQuantity;
+    }
+
+    /**
+     * The Method Object for handling {@link MakeSeatReservation} commands.
+     */
+    protected static class MakeSeatReservationCommandHandler {
```

Comment:
I've move this class to the level of the package, together with `Validator`. This aggregate is getting too big already.